MsCorePkg/Library/MemoryProtectionExceptionHandlerLib/MemoryProtectionExceptionHandlerLibAArch64.c

```diff
@@ -0,0 +1,483 @@
+    case 0x15: AbortCause = "Synchronous external abort on translation table walk, first level";
+      break;
+    case 0x16: AbortCause = "Synchronous external abort on translation table walk, second level";
+      break;
+    case 0x17: AbortCause = "Synchronous external abort on translation table walk, third level";
+      break;
+    case 0x1c: AbortCause = "Synchronous parity error on memory access on translation table walk, zeroth level";
+      break;
+    case 0x1d: AbortCause = "Synchronous parity error on memory access on translation table walk, first level";
+      break;
+    case 0x1e: AbortCause = "Synchronous parity error on memory access on translation table walk, second level";
+      break;
+    case 0x1f: AbortCause = "Synchronous parity error on memory access on translation table walk, third level";
+      break;
+    case 0x21: AbortCause = "Alignment fault";
+      break;
+    case 0x22: AbortCause = "Debug event";
+      break;
+    case 0x30: AbortCause = "TLB conflict abort";
+      break;
+    case 0x33:
+    case 0x34: AbortCause = "IMPLEMENTATION DEFINED";
+      break;
+    case 0x35:
+    case 0x36: AbortCause = "Domain fault";
+      break;
+    default: AbortCause = "";
+      break;
+  }
+
+  DEBUG ((DEBUG_ERROR, "\n%a: %a\n", AbortType, AbortCause));
+}
+
+STATIC
+VOID
+DescribeExceptionSyndrome (
+  IN UINT32  Esr
+  )
+{
+  CHAR8  *Message;
+  UINTN  Ec;
+  UINTN  Iss;
+
+  Ec  = Esr >> 26;
+  Iss = Esr & 0x00ffffff;
+
+  switch (Ec) {
+    case 0x15: Message = "SVC executed in AArch64";
+      break;
+    case 0x20:
+    case 0x21: DescribeInstructionOrDataAbort ("Instruction abort", Iss);
+      return;
+    case 0x22: Message = "PC alignment fault";
+      break;
+    case 0x23: Message = "SP alignment fault";
+      break;
+    case 0x24:
+    case 0x25: DescribeInstructionOrDataAbort ("Data abort", Iss);
+      return;
+    default: return;
+  }
+
+  DEBUG ((DEBUG_ERROR, "\n %a \n", Message));
+}
+
+#ifndef MDEPKG_NDEBUG
+STATIC
+CONST CHAR8 *
+BaseName (
+  IN  CONST CHAR8  *FullName
+  )
+{
+  CONST CHAR8  *Str;
+
+  Str = FullName + AsciiStrLen (FullName);
+
+  while (--Str > FullName) {
+    if ((*Str == '/') || (*Str == '\\')) {
+      return Str + 1;
+    }
+  }
+
+  return Str;
+}
+
+#endif
+
+/**
+  This is the default action to take on an unexpected exception
+
+  @param  ExceptionType    Type of the exception
+  @param  SystemContext    Register state at the time of the Exception
+
+**/
+VOID
+DefaultExceptionHandler (
+  IN     EFI_EXCEPTION_TYPE  ExceptionType,
+  IN OUT EFI_SYSTEM_CONTEXT  SystemContext
+  )
+{
+  CHAR8  Buffer[100];
+  UINTN  CharCount;
+  INT32  Offset;
+
+  DEBUG ((DEBUG_INFO, "%a:%d - ExceptionType: %d\n", __FUNCTION__, __LINE__, ExceptionType));
+  if (mRecursiveException) {
+    STATIC CHAR8 CONST  Message[] = "\nRecursive exception occurred while dumping the CPU state\n";
+
+    SerialPortWrite ((UINT8 *)Message, sizeof Message - 1);
+    if (gST->ConOut != NULL) {

```

Comment:
I assume this routine from ArmPkg? Some of the logics seems to need intricate service from the infrastructure (efi config tables, console, etc.), with the memory already corrupted, I am not sure how much of them would still work?

Reply:
It is the default exception handler from ArmPkg. Are you suggesting these prints be changed to DEBUG?

Reply:
Given we already dumped to serial port one line above, I think just remove this `AsciiPrint` attempt would be fine.

Reply:
Removed